### ISSUE TITLE
TypeError: %d format: a number is required, not BitVecAdd

### ISSUE THREAD

**Issue by suhabe** [trailofbits/manticore#813]
### OS / Environment
.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.3 LTS
Release:	16.04
Codename:	xenial

### Manticore version

Development version from git repo at commit eaa0e524b44cbdb4a7bcf3ba6808bbd86c2a16c3

### Python version

Python 2.7.12

### Dependencies

capstone==3.0.5rc2
manticore==0.1.7
ply==3.11
pyelftools==0.24
pysha3==1.0.2
unicorn==1.0.1
z3-solver==4.5.1.0.post2

### Summary of the problem
Exception thrown:
``TypeError: %d format: a number is required, not BitVecAdd``

### Step to reproduce the behavior
`manticore AttributeStore.sol`

```
pragma solidity ^0.4.8;

library AttributeStore {
    struct Data {
        mapping(bytes32 => uint) store;
    }

    function getAttribute(Data storage self, bytes32 _UUID, string _attrName) public view returns (uint) {
        bytes32 key = keccak256(_UUID, _attrName);
        return self.store[key];
    }

    function attachAttribute(Data storage self, bytes32 _UUID, string _attrName, uint _attrVal) public {
        bytes32 key = keccak256(_UUID, _attrName);
        self.store[key] = _attrVal;
    }
}
```

### Expected behavior
No exception thrown.

### Actual behavior
Exception thrown:
``TypeError: %d format: a number is required, not BitVecAdd``

### Any relevant logs
Manticore output is
```
$ manticore /home/vagrant/host/benchmarks/attribute_store.sol 
2018-03-15 21:07:50,254: [31783] m.main:INFO: Beginning analysis
2018-03-15 21:07:52,162: [31783] m.ethereum:INFO: Starting symbolic transaction: 1
2018-03-15 21:07:52,725: [31783] m.ethereum:WARNING: Integer overflow at ADD instruction
2018-03-15 21:07:52,729: [31783] m.c.executor:ERROR: Exception: %d format: a number is required, not BitVecAdd
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/executor.py", line 459, in run
    if not current_state.execute():
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/state.py", line 126, in execute
    result = self._platform.execute()
  File "/usr/local/lib/python2.7/dist-packages/manticore/platforms/evm.py", line 2208, in execute
    self.current.execute()
  File "/usr/local/lib/python2.7/dist-packages/manticore/platforms/evm.py", line 1373, in execute
    result = implementation(*arguments)
  File "/usr/local/lib/python2.7/dist-packages/manticore/platforms/evm.py", line 1625, in CALLDATALOAD
    bytes = list(self.data[offset:offset + 32])
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/smtlib/expression.py", line 718, in __getitem__
    new_array = ArrayVariable(self.index_bits, size, self.value_bits, name='%s_b%d_e%d'%(self.name, start, stop), taint=self.taint)
TypeError: %d format: a number is required, not BitVecAdd

2018-03-15 21:07:52,852: [31783] m.ethereum:INFO: Generated testcase No. 0 - REVERT
2018-03-15 21:07:53,455: [31783] m.c.executor:ERROR: Exception: AttributeStore.Data storage
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/executor.py", line 481, in run
    self.generate_testcase(current_state, str(e))
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/executor.py", line 353, in generate_testcase
    self._publish('will_generate_testcase', state, 'test', message)
  File "/usr/local/lib/python2.7/dist-packages/manticore/utils/event.py", line 118, in _publish
    self._publish_impl(_name, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/manticore/utils/event.py", line 134, in _publish_impl
    sink._publish_impl(_name, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/manticore/utils/event.py", line 126, in _publish_impl
    callback(robj(), *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/manticore/ethereum.py", line 1219, in _generate_testcase_callback
    function_name, arguments = ABI.parse(signature, tx.data)
  File "/usr/local/lib/python2.7/dist-packages/manticore/ethereum.py", line 435, in parse
    val, off = ABI._consume_type(ty, data, off)
  File "/usr/local/lib/python2.7/dist-packages/manticore/ethereum.py", line 413, in _consume_type
    raise NotImplementedError(ty)
NotImplementedError: AttributeStore.Data storage

2018-03-15 21:07:53,479: [31783] m.ethereum:INFO: Generated testcase No. 1 - REVERT
2018-03-15 21:07:54,093: [31783] m.ethereum:INFO: Finished symbolic transaction: 1 | Code Coverage: 14% | Terminated States: 3 | Alive States: 2
2018-03-15 21:07:54,099: [31783] m.ethereum:INFO: Starting symbolic transaction: 2
^C2018-03-15 21:07:54,134: [31783] m.c.executor:ERROR: Exception: 
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/executor.py", line 459, in run
    if not current_state.execute():
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/state.py", line 126, in execute
    result = self._platform.execute()
  File "/usr/local/lib/python2.7/dist-packages/manticore/platforms/evm.py", line 2204, in execute
    self._process_pending_transaction()
  File "/usr/local/lib/python2.7/dist-packages/manticore/platforms/evm.py", line 2367, in _process_pending_transaction
    enough_balance_solutions = solver.get_all_values(self._constraints, enough_balance)
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/smtlib/solver.py", line 356, in get_all_values
    while self._check() == 'sat':
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/smtlib/solver.py", line 271, in _check
    raise SolverException(_status)
SolverException

2018-03-15 21:07:54,151: [31783] m.c.executor:ERROR: Exception: [Errno 32] Broken pipe
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/executor.py", line 492, in run
    if solver.check(current_state.constraints):
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/smtlib/solver.py", line 66, in check
    return self.can_be_true(constraints, True)
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/smtlib/solver.py", line 331, in can_be_true
    self._reset(temp_cs.related_to(expression))
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/smtlib/solver.py", line 217, in _reset
    self._send("(reset)")
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/smtlib/solver.py", line 237, in _send
    raise SolverException(e)
SolverException: [Errno 32] Broken pipe

2018-03-15 21:07:54,179: [31783] m.ethereum:INFO: Finished symbolic transaction: 2 | Code Coverage: 14% | Terminated States: 3 | Alive States: 2
2018-03-15 21:07:54,214: [32520] m.ethereum:INFO: Generated testcase No. 2 - 
Process Process-3:
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python2.7/dist-packages/manticore/ethereum.py", line 1285, in f
    self.terminate_state_id(state_id)
  File "/usr/local/lib/python2.7/dist-packages/manticore/ethereum.py", line 752, in terminate_state_id
    self._generate_testcase_callback(state, 'test', last_exc.message)
  File "/usr/local/lib/python2.7/dist-packages/manticore/ethereum.py", line 1166, in _generate_testcase_callback
    balance = state.solve_one(balance)
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/state.py", line 313, in solve_one
    return self._solver.get_value(self._constraints, expr)
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/smtlib/solver.py", line 484, in get_value
    self._reset(temp_cs)
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/smtlib/solver.py", line 217, in _reset
    self._send("(reset)")
  File "/usr/local/lib/python2.7/dist-packages/manticore/core/smtlib/solver.py", line 237, in _send
    raise SolverException(e)
SolverException: [Errno 32] Broken pipe
2018-03-15 21:07:54,233: [31783] m.ethereum:INFO: Results in /home/vagrant/host/manticore/mcore_IXeDgW
```

**Comment by mossberg** [comment]
hi @suhabe, sorry that you ran into this bug, thanks a lot for filing this issue! :)

**Comment by defunctio** [comment]
Closing this issue as it appears to be resolved in the current master 1f74f0fa84a1aa3911c68df1b9cf20800dd7c778; however this example did happen trigger #1005.
